dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/GitHubSourceCodeRepo.java

```diff
@@ -333,7 +336,17 @@ private String readFileFromRepo(final String originalFileName, final String orig
             if (decodedContentAndMetadata == null) {
                 return null;
             } else {
-                return decodedContentAndMetadata.getRight();
+                String content = decodedContentAndMetadata.getRight();
+                String encoding = decodedContentAndMetadata.getLeft().getEncoding();
+                // If the file size is 1MB or larger, content will be "" and the encoding will be "none":
+                // https://docs.github.com/en/rest/repos/contents?apiVersion=2022-11-28 (see "Notes")
+                // In such a case, we retrieve the content via the blob endpoint.
+                if ("".equals(content) && "none".equals(encoding)) {
+                    String sha = decodedContentAndMetadata.getLeft().getSha();
+                    GHBlob blob = repo.getBlob(sha);
+                    content = IOUtils.toString(blob.read(), StandardCharsets.UTF_8);
```

Comment:
According to [this](https://docs.github.com/en/rest/git/blobs?apiVersion=2022-11-28#get-a-blob), a blob can be up to 100MB. And we duplicate it, so that will be 200MB of memory.  I think we can handle many of these at the same time, but still, I wonder if we could be more proactive. Also, I know it's unlikely, but somebody could craft a repo/dockstore.yml that might DOS us.

Can you tell what the size is without fetching it?

Reply:
The code checks if the size of the blob is more than `MAXIMUM_FILE_DOWNLOAD_SIZE`, and if it is, it doesn't download it.